fmriprep/workflows/bold/fit.py

```diff
@@ -211,10 +212,15 @@ def init_bold_fit_wf(
     # Collect sbref files, sorted by EchoTime
     sbref_files = get_sbrefs(
         bold_series,
-        entity_overrides=config.execution.get().get('bids_filters', {}).get('sbref', {}),
+        entity_overrides=bids_filters.get('sbref', {}),
         layout=layout,
     )
 
+    # T2w is not required, but if available, use to improve BOLD -> anat coreg
+    has_t2w = bool(
+        layout.get(suffix='T2w', extension=['.nii', '.nii.gz'], **bids_filters.get('t2w', {}))
+    )
```

Comment:
I would rather that we determine `has_t2w` (or whatever) at the `workflows.base` and pass it up. That will allow us to check whether `t2w` appears in the anatomical directory or `t2w_preproc` appears in precomputed derivatives. Hitting the layout and re-applying filters in here feels fragile.